### PR TITLE
Support patching other Go plugins

### LINKED ISSUE #21: use of external plugins breaks generated code
We often use other plugins like `protoc-gen-validate` or `protoc-gen-grpc-gateway`, but tools renaming fields usually breaks the generated code. The chosen approach by `protopatch` may solve this issue but does not work right now.
I did implement a [naive cache](https://github.com/Adphi/protopatch/tree/cache) that store `protoc-gen-go` output and reload it before patching generated code by other plugins, allowing the ast tree to find the type's declarations. I can create a pull request if you wish.
Another approach would have been to re-run `protoc-gen-go` plugin before each (other) plugin call to retrieve the types declaration.
I don't know which would be best.
Any thought ?

### PR DESCRIPTION
**Pull request by ydnar**
Fixes #21 by running the proto → Go code generation that normally executes in `protoc-gen-go` internally so the types and symbols it generates are available for type checking and patching other generated code.

See this repo for code that currently breaks: https://github.com/Adphi/protopatch-uncache-test

### CODE REVIEW

**Comment by ydnar** [comment]
In its current state, this PR fixes half of the issues in #21. The real fix is to change how it synthesizes unknown types. The Go type checker looks at type switches and fails if the field or variable being switched on is not an interface type.

The fix is to synthesize structs, fields, and methods separately.

**Comment by ydnar** [comment]
@Adphi mind checking this out? It should fix the issues you were seeing in #21.

File: patch/patcher.go [hunk 1]
```diff
@@ -75,6 +75,13 @@ func (p *Patcher) scan() error {
 func (p *Patcher) scanFile(f *protogen.File) {
 	log.Printf("\nScan proto:\t%s", f.Desc.Path())
 
+	// Locally generate Go from the source proto file.
```
Comment (ydnar) [inline patch/patcher.go:78]: This is the critical part of this PR. It uses the `internal_gengo` package to run `protoc-gen-go` in-process.

**Comment by Adphi** [comment]
@ydnar Interesting. 

I'll take a look in the next few days.

**Comment by ydnar** [comment]
@Adphi thanks for the example repo! Is it OK that I copied it into this PR as a smoke test for `protoc-gen-validate`?

**Comment by ydnar** [comment]
@Adphi if this works for your use case, I’m going to merge to `main`. I want to help rebase #22 and #15 since this PR will create a few merge conflicts/

**Comment by Adphi** [comment]
@ydnar I works for me, thanks.

> @Adphi thanks for the example repo! Is it OK that I copied it into this PR as a smoke test for `protoc-gen-validate`?

Of course !

**Pull request merged**